markers/sauce.py

```diff
@@ -14,7 +17,31 @@ def pytest_addoption(parser):
 def pytest_configure(config):
     config.addinivalue_line('markers', __doc__.splitlines()[0])
     if config.option.sauce:
+        conf.env['browser']['sauce'] = True
         if config.option.markexpr:
             config.option.markexpr = 'sauce and ({})'.format(config.option.markexpr)
         else:
             config.option.markexpr = 'sauce'
+
+
+def pytest_runtest_setup(item):
+    browser_name = None
+    if 'chrome' in str(item).lower():
```

Comment:
This is not the right way to get the parameter, we can do this in a much better way, I will get those details tomorrow because they escape me now, but this has the potential to get ugly and unpredictable relying on the string matching. I mean it _should_ be safe, but I think we should do it properly.